peachjam/views/gazette.py

```diff
@@ -36,13 +37,27 @@ class GazetteListView(TemplateView):
     template_name = "peachjam/gazette_list.html"
     navbar_link = "gazettes"
 
+    def get(self, request, code=None, *args, **kwargs):
+        self.locality = get_object_or_404(Locality, code=code) if code else None
+        return super().get(request, *args, **kwargs)
+
     def get_queryset(self):
-        return self.queryset
+        qs = self.queryset
+        if self.locality:
+            qs = qs.filter(locality=self.locality)
+        return qs
 
     def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
+        context = super().get_context_data(locality=self.locality, **kwargs)
 
         queryset = self.get_queryset()
+        localities = list(Locality.objects.all())
+        gazettes = queryset.filter(locality__in=localities)
```

Comment:
Consider this in context. When viewing South Africa, this loads 60,000 gazettes and iterates over all of them, just to add 9 localities to the set. It's very inefficient. It's also going to be doing it on a page where locality is set, even though it's never used.

Here's a better query:

```
>>> str(qs.order_by().distinct("locality").values_list("locality", flat=True).query)
'SELECT DISTINCT ON ("peachjam_coredocument"."locality_id") "peachjam_coredocument"."locality_id" FROM "peachjam_gazette" INNER JOIN "peachjam_coredocument" ON ("peachjam_gazette"."coredocument_ptr_id" = "peachjam_coredocument"."id")'
>>> list(qs.order_by().distinct("locality").values_list("locality", flat=True))
[6, None]
```

So instead,

1. only calculate this when `self.locality` is None
2. use this to get distinct locality ids: `list(qs.order_by().distinct("locality").values_list("locality", flat=True))`